iOS/Components/Browser/Album Detail/AlbumSongListViewController.swift

```diff
@@ -0,0 +1,145 @@
+//
+//  AlbumSongListViewController.swift
+//  Persephone
+//
+//  Created by Daniel Barber on 2020-3-30.
+//  Copyright © 2020 Dan Barber. All rights reserved.
+//
+
+import UIKit
+import Kingfisher
+
+class AlbumSongListViewController: UITableViewController {
+  var album: Album?
+  var albumSongs: [Song] = []
+  var dataSource = AlbumTracksDataSource()
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    
+    albumTracksView.dataSource = dataSource
+    albumTracksView.layoutMargins = UIEdgeInsets(top: 0, left: 20, bottom: 0, right: 20)
+    
+    navigationItem.largeTitleDisplayMode = .never
+    
+    albumCoverView.layer.backgroundColor = UIColor.black.cgColor
+    albumCoverView.layer.cornerRadius = 4
+    albumCoverView.layer.borderWidth = 0.5
+    albumCoverView.layer.masksToBounds = true
+
+    playAlbumButton.layer.cornerRadius = 8
+    addAlbumToQueueButton.layer.cornerRadius = 8
+    setAppearance()
+  }
+
+  override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+
+    guard let album = album else { return }
+    
+    getAlbumSongs(for: album)
+
+    albumTitle.text = album.title
+    albumMetadata.text = "\(album.artist) · \(album.date)"
+
+    var layoutSize = UIView.layoutFittingExpandedSize
+    layoutSize.width = UIScreen.main.bounds.width
```

Comment:
FWIW, a good place to work out the layout width is in `viewWillLayoutSubviews()` or `viewDidLayoutSubviews()`. Moving this code there and using `view.bounds.width` instead of `UIScreen.main.bounds.width` will make this more robust to things like orientation changes or size changes on iPad.

Reply:
I moved it here because it was updating the layout often and I was noticing some jankyness with the calculation (the sizes of things would jump about). Now I've figured this it out I can probably move it back.

Reply:
Something I also usually do here is to cache the previous width, so the layout only needs to be recalculated if the width has changed.